draft-sharabayko-srt.md

```diff
@@ -674,11 +674,33 @@ The purpose of the extension is to let peers exchange and negotiate encryption-r
 to be used to encrypt and decrypt the payload of the stream.
 
 The extension can be supplied with the Handshake Extension Type field set
-to either SRT_CMD_KMREQ or SRT_CMD_HSRSP (see {{handshake-ext-type}} in {{ctrl-pkt-handshake}}).
+to either SRT_CMD_KMREQ or SRT_CMD_KMRSP (see {{handshake-ext-type}} in {{ctrl-pkt-handshake}}).
 For more details refer to {{handshake-messages}}.
 
 The KM message is placed in the Extension Contents. See {{sec-ctrlpkt-km}} for the structure of the KM message.
 
+In case of the SRT_CMD_KMRSP the Extension Length value can be equal to 1 (meaning 4 bytes).
+It is an indication of encryption failure. In this case the Extension Content has a different format, {{fig-hsext-kmerror}}.
+
+~~~
+ 0                   1                   2                   3
+ 0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1
++-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+|                           KM State                            |
++-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+~~~
+{: #fig-hsext-kmerror title="KM Response Error"}
+
+KM State: 32 bits.
+: Key Material State of the peer:
+
+- 0: unsecured (the peer will encrypt the payload, while the agent has not declared any encryption),
+- 3: no secret (the peer does not have a key to decrypt the incoming payload),
+- 4: bad secret (the peer has a wrong key and can't decrypt the incoming payload),
+- 5: bad crypto mode (the peer expects to use a different cryptographic mode). Since protocol v1.6.
+
+Here "peer" indicates the remove SRT side sending the KM response, and "agent" means the local side interpreting the KM response.
```

Comment:
```suggestion
NOTE: In the descriptions below, "peer" refers to the remote SRT side sending the KM response, and "agent" refers to the local side interpreting the KM response.

- 0: unsecured (the peer will encrypt the payload, while the agent has not declared any encryption),
- 3: no secret (the peer does not have the key to decrypt the incoming payload),
- 4: bad secret (the peer has the wrong key and can't decrypt the incoming payload),
- 5: bad crypto mode (the peer expects to use a different cryptographic mode). Since protocol v1.6.


```